tests/db_functions/math/test_atan.py

```diff
@@ -0,0 +1,44 @@
+import math
+from decimal import Decimal
+
+from django.db.models import DecimalField
+from django.db.models.functions import ATan
+from django.test import TestCase
+
+from ..models import DecimalModel, FloatModel, IntegerModel
+
+
+class ATanTests(TestCase):
+
+    def test_decimal(self):
+        DecimalModel.objects.create(n1=Decimal('-12.9'), n2=Decimal('0.6'))
+        obj = DecimalModel.objects.annotate(n1_atan=ATan('n1'), n2_atan=ATan('n2')).first()
+        self.assertAlmostEqual(obj.n1_atan, math.atan(obj.n1))
+        self.assertAlmostEqual(obj.n2_atan, math.atan(obj.n2))
+
+    def test_float(self):
+        FloatModel.objects.create(f1=-27.5, f2=0.33)
+        obj = FloatModel.objects.annotate(f1_atan=ATan('f1'), f2_atan=ATan('f2')).first()
+        self.assertAlmostEqual(obj.f1_atan, math.atan(obj.f1))
+        self.assertAlmostEqual(obj.f2_atan, math.atan(obj.f2))
+
+    def test_integer(self):
+        IntegerModel.objects.create(small=-20, normal=15, big=-1)
+        obj = IntegerModel.objects.annotate(
+            small_atan=ATan('small'),
+            normal_atan=ATan('normal'),
+            big_atan=ATan('big'),
+        ).first()
+        self.assertAlmostEqual(obj.small_atan, math.atan(obj.small))
+        self.assertAlmostEqual(obj.normal_atan, math.atan(obj.normal))
+        self.assertAlmostEqual(obj.big_atan, math.atan(obj.big))
+
+    def test_transform(self):
+        try:
+            DecimalField.register_lookup(ATan)
+            DecimalModel.objects.create(n1=Decimal('3.12'), n2=Decimal('0'))
+            DecimalModel.objects.create(n1=Decimal('-5'), n2=Decimal('0'))
+            objs = DecimalModel.objects.filter(n1__atan__gt=0)
+            self.assertQuerysetEqual(objs, [3.12], lambda a: float(a.n1))
```

Comment:
Simplify to `lambda a: a.n1`

Reply:
That's gonna give me error message `AssertionError: Lists differ: [Decimal('3.12')] != [3.12]`

Reply:
Instead, I changed it to `self.assertQuerysetEqual(objs, [Decimal('3.12')], lambda a: a.n1)`.
Problem solved.

Reply:
Yes. That's what it should be.